scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -447,8 +429,13 @@ object Ctor {
 
 @ast class Self(name: Name, decltpe: Option[Type]) extends Member
 
-@ast class Template(early: List[Stat], inits: List[Init], self: Self, stats: List[Stat])
-    extends Tree {
+@ast class Template(
+    early: List[Stat],
+    inits: List[Init],
+    derives: List[Term.Name],
```

Comment:
I would try to add derives an additional private var and expose it via additional apply/unapply method maybe? Or we could just add an additional unapply method - that should keep the compatibility, no?

Reply:
I can try, there is this `@ast` macro that I have feeling might not cope with this idea, but we will see

Reply:
in macro:
```
illegal.foreach(stmt =>
          c.abort(
            stmt.pos,
            "only checkFields(...), checkParent(...) and definitions are allowed in @ast classes"
          )
        )
```

Reply:
reverted to not include `derives` in this PR.